databuilder/databuilder/models/dashboard/dashboard_usage.py

```diff
@@ -51,73 +41,29 @@ def __init__(self,
-        self._record_iterator = self._create_record_iterator()
-
-    def create_next_node(self) -> Union[GraphNode, None]:
-        if self._should_create_user_node:
-            return self._user_model.create_next_node()
-
-        return None
-
-    def create_next_relation(self) -> Union[GraphRelationship, None]:
-        try:
-            return next(self._relation_iterator)
-        except StopIteration:
-            return None
-
-    def _create_relation_iterator(self) -> Iterator[GraphRelationship]:
-        relationship = GraphRelationship(
+        Usage.__init__(
+            self,
             start_label=DashboardMetadata.DASHBOARD_NODE_LABEL,
-            end_label=User.USER_NODE_LABEL,
             start_key=DashboardMetadata.DASHBOARD_KEY_FORMAT.format(
-                product=self._product,
-                cluster=self._cluster,
-                dashboard_group=self._dashboard_group_id,
-                dashboard_name=self._dashboard_id
+                product=product,
+                cluster=cluster,
+                dashboard_group=dashboard_group_id,
+                dashboard_name=dashboard_id
             ),
-            end_key=User.get_user_model_key(email=self._email),
-            type=READ_REVERSE_RELATION_TYPE,
-            reverse_type=READ_RELATION_TYPE,
-            attributes={
-                READ_RELATION_COUNT_PROPERTY: self._view_count
-            }
+            user_email=email,
+            read_count=view_count,
         )
-        yield relationship
-
-    def create_next_record(self) -> Union[RDSModel, None]:
-        try:
-            return next(self._record_iterator)
-        except StopIteration:
-            return None
 
-    def _create_record_iterator(self) -> Iterator[RDSModel]:
+    # override superclass for customized _should_create_user_node behavior
+    def create_next_node(self) -> Union[GraphNode, None]:
         if self._should_create_user_node:
-            user_record = self._user_model.create_next_record()
-            if user_record:
-                yield user_record
-
-        dashboard_usage_record = RDSDashboardUsage(
-            user_rk=User.get_user_model_key(email=self._email),
-            dashboard_rk=DashboardMetadata.DASHBOARD_KEY_FORMAT.format(
-                product=self._product,
-                cluster=self._cluster,
-                dashboard_group=self._dashboard_group_id,
-                dashboard_name=self._dashboard_id
-            ),
-            read_count=self._view_count
-        )
-        yield dashboard_usage_record
+            return super().create_next_node()
+        return None
 
-    def __repr__(self) -> str:
-        return f'DashboardUsage({self._dashboard_group_id!r}, {self._dashboard_id!r}, ' \
-               f'{self._email!r}, {self._view_count!r}, {self._should_create_user_node!r}, ' \
-               f'{self._product!r}, {self._cluster!r})'
+    # override superclass for customized _should_create_user_node behavior
```

Comment:
nit: what do you think if we move `_should_create_user_node`(defaults to true) to Usage constructor? Thanks

Reply:
This is a good idea, and would also be helpful in the generic Owner class, which has to do the same thing. If you don't mind, I will do this in a follow up.